pkg/resmgr/control/cpu/cpu.go

```diff
@@ -271,21 +263,17 @@ func (ctl *cpuctl) configure(cfg *cfgapi.Config) error {
 
 	// Configure the system
 	for class, cpus := range assignments {
-		if _, ok := ctl.classes[class]; ok {
-			// Re-configure cpus (sysfs) according to new class parameters
-			if err := ctl.enforceCpufreq(class, cpus.SortedMembers()...); err != nil {
-				log.Error("cpufreq enforcement on re-configure failed: %v", err)
-			}
-			if err := ctl.enforceCpufreqGovernor(class, cpus.SortedMembers()...); err != nil {
-				log.Error("cpufreq enforcement on re-configure failed: %v", err)
-			}
-		} else {
+		if _, ok := ctl.classes[class]; !ok {
 			// TODO: what should we really do with classes that do not exist in
 			// the configuration anymore? Now we remember the CPUs assigned to
 			// them. A further config update might re-introduce the class in
 			// which case the CPUs will be reconfigured.
 			log.Warn("class %q with cpus %v missing from the configuration", class, cpus)
 		}
```

Comment:
If you flip the if-ok-else branch to if-!ok, you either need to use a if-!ok-else construct (IOW, move enforceCpufreq into an else block) or add a continue here in the if block. Otherwise we'll log a warning for a non-existent class but still try to enforce CPU frequencies and the governor for it...